test/lib/cql_test_env.cc

```diff
@@ -249,8 +253,10 @@ class single_node_cql_env : public cql_test_env {
 
         auto qs = make_query_state();
         auto& lqo = *qo;
-        return local_qp().execute_prepared(std::move(prepared), std::move(id), *qs, lqo, true)
-            .finally([qs, qo = std::move(qo)] {});
+        return local_qp().execute_prepared_without_checking_exception_message(std::move(prepared), std::move(id), *qs, lqo, true)
+            .then([qs, qo = std::move(qo)] (auto msg) {
+                return cql_transport::messages::propagate_exception_as_future(std::move(msg));
```

Comment:
But this is just the testing environment... How is it relevant to Scylla performance? I guess you are running your test in the testing environment instead of real Scylla?

Reply:
I meant the performance of the `perf_simple_query` benchmark. Yes, this is just testing code and does not have impact on the real Scylla, so maybe the regression I measured is completely virtual (on the other hand maybe there is some undetected regression outside the code path tested by the benchmark...).